### PR TITLE
1.12.2: Restructuring events & some fixes

### PR DESCRIPTION
**Pull request by sora7672**
The small fixes are from other discord members or i found while building my example repo for 1.12.2.
They were from 5th-11th april.
After that i basically changed all event pages, so they have the same buildup.
(because i worked sometimes to long, i created some errors which i fixed later on, thats why sometimes 3 chanegs are at the same file)

With this it should be easyer to follow for no programmers to follow where they find stuff.

I hope its fine for you like this too :)

### CODE REVIEW

File: docs/1.12/content/Vanilla/Events/Events/PlayerSetSpawn.md [hunk 1]
```diff
@@ -14,11 +14,28 @@ PlayerSetSpawn Events implement the following interfaces and are able to call al
...
+## ZenGetters/ZenSetters/ZenMethods from extensions
+The following information can be retrieved from the event:
+| ZenGetter       | ZenSetter       | Type                                              |
+|-----------------|-----------------|---------------------------------------------------|
+| `player`         |                 | [IPlayer](/Vanilla/Players/IPlayer/)             |
+| `canceled`      | `canceled`      | bool                                              |
+
+
+ZenMethods
+- `event.cancel();` Method, returns void (nothing). Can cancel the event and stop smth. from happening
```
Comment (jaredlll08) [inline docs/1.12/content/Vanilla/Events/Events/PlayerSetSpawn.md:34]:
```suggestion
- `event.cancel();` Method, returns void (nothing). Can cancel the event and stop something from happening
```

File: docs/1.12/content/Vanilla/Events/Events/PlayerUseHoe.md [hunk 2]
```diff
@@ -15,25 +15,55 @@ PlayerUseHoe Events implement the following interfaces and are able to call all
...
+| `player`        |                 | [IPlayer](/Vanilla/Players/IPlayer/)              |
+| `result`        |                 | string with value of `default`, `deny` or `allow` |
+| `canceled`      | `canceled`      | bool                                              |
+
+
+ZenMethods
+- `event.deny()` Method, sets the event's result to `deny`
+- `event.allow()` Method, sets the event's result to `allow`
+- `event.default()` Method, sets the event's result to `default`
+- `event.cancel();` Method, returns void (nothing). Can cancel the event and stop smth. from happening
```
Comment (jaredlll08) [inline docs/1.12/content/Vanilla/Events/Events/PlayerUseHoe.md:58]:
```suggestion
- `event.cancel();` Method, returns void (nothing). Can cancel the event and stop something from happening
```

File: docs/1.12/content/Vanilla/Events/Events/PlayerUseItemStart.md [hunk 3]
```diff
@@ -14,15 +14,27 @@ PlayerUseItemStart Events implement the following interfaces and are able to cal
...
-- `event.cancel()` sets the event as cancelled
\ No newline at end of file
+| ZenGetter       | ZenSetter       | Type                                              |
+|-----------------|-----------------|---------------------------------------------------|
+| `canceled`      | `canceled`      | bool                                              |
+| `player`        |                 | [IPlayer](/Vanilla/Players/IPlayer/)              |
+
+
+ZenMethods
+- `event.cancel();` Method, returns void (nothing). Can cancel the event and stop smth. from happening
```
Comment (jaredlll08) [inline docs/1.12/content/Vanilla/Events/Events/PlayerUseItemStart.md:34]:
```suggestion
- `event.cancel();` Method, returns void (nothing). Can cancel the event and stop something from happening
```

File: docs/1.12/content/Vanilla/Events/Events/PlayerUseItemTick.md [hunk 4]
```diff
@@ -14,16 +14,29 @@ PlayerUseItemTick Events implement the following interfaces and are able to call
...
+The following information can be retrieved/set during the event:
+
+| ZenGetter       | ZenSetter       | Type                                              |
+|-----------------|-----------------|---------------------------------------------------|
+| `canceled`      | `canceled`      | bool                                              |
+| `player`        |                 | [IPlayer](/Vanilla/Players/IPlayer/)              |
+
+
+ZenMethods
+- `event.cancel();` Method, returns void (nothing). Can cancel the event and stop smth. from happening
```
Comment (jaredlll08) [inline docs/1.12/content/Vanilla/Events/Events/PlayerUseItemTick.md:36]:
```suggestion
- `event.cancel();` Method, returns void (nothing). Can cancel the event and stop something from happening
```

File: docs/1.12/content/Vanilla/Events/Events/PortalSpawn.md [hunk 5]
```diff
@@ -15,10 +15,41 @@ BlockNeighborNotify Events implement the following interfaces and are able to ca
...
+| ZenGetter       | ZenSetter       | Type                                              |
+|-----------------|-----------------|---------------------------------------------------|
+| `world`         |                 | [IWorld](/Vanilla/World/IWorld/)                  |
+| `blockState`    |                 | [IBlockState](/Vanilla/Blocks/IBlockState/)       |
+| `block`         |                 | [IBlock](/Vanilla/Blocks/IBlock/)                 |
+| `canceled`      | `canceled`      | bool                                              |
+
+
+ZenMethods
+- `event.cancel();` Method, returns void (nothing). Can cancel the event and stop smth. from happening
```
Comment (jaredlll08) [inline docs/1.12/content/Vanilla/Events/Events/PortalSpawn.md:42]:
```suggestion
- `event.cancel();` Method, returns void (nothing). Can cancel the event and stop something from happening
```

File: docs/1.12/content/Vanilla/Events/Events/PotionBrewPre.md [hunk 6]
```diff
@@ -14,3 +12,21 @@ PotionBrewPre Events implement the following interfaces and are able to call all
...
+The following information can be retrieved/set during the event:
+
+| ZenGetter       | ZenSetter       | Type                                              |
+|-----------------|-----------------|---------------------------------------------------|
+| `length`        |                 | int                                               |
+| `canceled`      | `canceled`      | bool                                              |
+
+
+ZenMethods
+- `event.cancel();` Method, returns void (nothing). Can cancel the event and stop smth. from happening
```
Comment (jaredlll08) [inline docs/1.12/content/Vanilla/Events/Events/PotionBrewPre.md:27]:
```suggestion
- `event.cancel();` Method, returns void (nothing). Can cancel the event and stop something from happening
```

File: docs/1.12/content/Vanilla/Events/Events/ProjectileImpactArrow.md [hunk 7]
```diff
@@ -27,16 +27,26 @@ The following information can be retrieved from the event:
...
 
-- `setPickupAllowed()`
+| ZenGetter       | ZenSetter       | Type                                              |
+|-----------------|-----------------|---------------------------------------------------|
+| `canceled`      | `canceled`      | bool                                              |
+| `rayTrace`      |                 | [IRayTraceResult](/Vanilla/World/IRayTraceResult/) |
 
-Allows the arrow to be picked up from where the entity landed.
+ZenMethods
+- `event.cancel();` Method, returns void (nothing). Can cancel the event and stop smth. from happening
```
Comment (jaredlll08) [inline docs/1.12/content/Vanilla/Events/Events/ProjectileImpactArrow.md:45]:
```suggestion
- `event.cancel();` Method, returns void (nothing). Can cancel the event and stop something from happening
```

File: docs/1.12/content/Vanilla/Events/Events/ProjectileImpactFireball.md [hunk 8]
```diff
@@ -25,3 +25,21 @@ The following information can be retrieved from the event:
...
+## ZenGetters/ZenSetters/ZenMethods from extensions
+The following information can be retrieved/set during the event:
+
+| ZenGetter       | ZenSetter       | Type                                              |
+|-----------------|-----------------|---------------------------------------------------|
+| `canceled`      | `canceled`      | bool                                              |
+| `rayTrace`      |                 | [IRayTraceResult](/Vanilla/World/IRayTraceResult/) |
+
+ZenMethods
+- `event.cancel();` Method, returns void (nothing). Can cancel the event and stop smth. from happening
```
Comment (jaredlll08) [inline docs/1.12/content/Vanilla/Events/Events/ProjectileImpactFireball.md:38]:
```suggestion
- `event.cancel();` Method, returns void (nothing). Can cancel the event and stop something from happening
```

File: docs/1.12/content/Vanilla/Events/Events/ProjectileImpactThrowable.md [hunk 9]
```diff
@@ -6,17 +6,34 @@ This event is fired whenever a throwable projectile impacts an entity but before
...
+## ZenGetters/ZenSetters/ZenMethods from extensions
+The following information can be retrieved/set during the event:
+
+| ZenGetter       | ZenSetter       | Type                                              |
+|-----------------|-----------------|---------------------------------------------------|
+| `canceled`      | `canceled`      | bool                                              |
+| `rayTrace`      |                 | [IRayTraceResult](/Vanilla/World/IRayTraceResult/) |
+
+ZenMethods
+- `event.cancel();` Method, returns void (nothing). Can cancel the event and stop smth. from happening
```
Comment (jaredlll08) [inline docs/1.12/content/Vanilla/Events/Events/ProjectileImpactThrowable.md:32]:
```suggestion
- `event.cancel();` Method, returns void (nothing). Can cancel the event and stop something from happening
```

File: docs/1.12/content/Vanilla/Events/Events/PlayerCrafted.md [hunk 10]
```diff
@@ -1,6 +1,7 @@
 # PlayerCrafted
 
 The PlayerCrafted Event is fired whenever a player crafts something.
+Which to be more precise, the player takes out the result, not changing smth. in the grid.
```
Comment (jaredlll08) [inline docs/1.12/content/Vanilla/Events/Events/PlayerCrafted.md:4]:
```suggestion
Which to be more precise, the player takes out the result, not changing something in the grid.
```

File: docs/1.12/content/Vanilla/Events/Events/CriticalHit.md [hunk 11]
```diff
@@ -37,3 +37,9 @@ The following information can be retrieved from the event:
 - `event.deny()` sets the result to deny.
 - `event.allow()` sets the result to allow.
 - `event.default()` sets the result to default.
+
+## ZenGetters/ZenSetters/ZenMethods from extensions
+
+| ZenGetter       | ZenSetter       | Type                                              |
+|-----------------|-----------------|---------------------------------------------------|
+| `player`        |                 | [IBlockPos](/Vanilla/Player/IPlayer/)             |
```
Comment (jaredlll08) [inline docs/1.12/content/Vanilla/Events/Events/CriticalHit.md:45]:
```suggestion
| `player`        |                 | [IPlayer](/Vanilla/Player/IPlayer/)             |
```

File: docs/1.12/content/Vanilla/Events/Events/PlayerSleepInBed.md [hunk 12]
```diff
@@ -15,30 +15,29 @@ PlayerSleepInBed Events implement the following interfaces and are able to call
...
+
+## Extension from extension
+The following information can be retrieved/set during the event:
+| ZenGetter       | ZenSetter       | Type                                              |
+|-----------------|-----------------|---------------------------------------------------|
+| `entityLivingBase`  |             | [IEntityLivingBase](/Vanilla/Entities/IEntityLivingBase/)   |
+
+## Notes
+`result` can return or set the strings:
+`NOT_POSSIBLE_HERE`, `NOT_POSSIBLE_NOW`, `NOT_SAFE`, `OK` or  `OTHER_PROBLEM`
```
Comment (jaredlll08) [inline docs/1.12/content/Vanilla/Events/Events/PlayerSleepInBed.md:43]:
```suggestion
`NOT_POSSIBLE_HERE`, `NOT_POSSIBLE_NOW`, `NOT_SAFE`, `OK` or `OTHER_PROBLEM`
```

File: docs/1.12/content/Vanilla/Events/Events/PlayerRightClickItem.md [hunk 13]
```diff
@@ -14,10 +14,35 @@ PlayerRightClickItem Events implement the following interfaces and are able to c
 
 - [IEventCancelable](/Vanilla/Events/Events/IEventCancelable/)
 - [PlayerInteract](/Vanilla/Events/Events/PlayerInteract/)
+- [IHasCancellationResult] (no doc page so far)
```
Comment (jaredlll08) [inline docs/1.12/content/Vanilla/Events/Events/PlayerRightClickItem.md:17]:
```suggestion
- IHasCancellationResult
```
If it doesn't have a page, rather don't link, compared to linking to a 404

File: docs/1.12/content/Vanilla/Events/Events/IPlayerEvent.md [hunk 14]
```diff
@@ -2,6 +2,9 @@
 This interface is extended by all events that have a player.  
 That means you can use the getter below to access the player. 
 
+## Note
+The  [IPlayer](/Vanilla/Players/IPlayer/)  is an extension of the class [IEntityLivingBase](/Vanilla/Entities/IEntityLivingBase/) ,
```
Comment (jaredlll08) [inline docs/1.12/content/Vanilla/Events/Events/IPlayerEvent.md:6]:
```suggestion
The [IPlayer](/Vanilla/Players/IPlayer/) is an extension of the class [IEntityLivingBase](/Vanilla/Entities/IEntityLivingBase/) ,
```

File: docs/1.12/content/Vanilla/Events/Events/IPlayerEvent.md [hunk 15]
```diff
@@ -2,6 +2,9 @@
...
+thus with `event.player` you have access to all  ZenGetters/ZenSetters from [IEntityLivingBase](/Vanilla/Entities/IEntityLivingBase/) allready.
```
Comment (jaredlll08) [inline docs/1.12/content/Vanilla/Events/Events/IPlayerEvent.md:7]:
```suggestion
thus with `event.player` you have access to all  ZenGetters/ZenSetters from [IEntityLivingBase](/Vanilla/Entities/IEntityLivingBase/) already.
```

File: docs/1.12/content/Vanilla/Events/Events/PlayerDeathDrops.md [hunk 16]
```diff
@@ -13,14 +13,26 @@ PlayerDeathDrops Events implement the following interfaces and are able to call
...
 The following information can be retrieved from the event:
 
-| ZenGetter      | Return Type                                          |
-|----------------|------------------------------------------------------|
-| `player`       | [IPlayer](/Vanilla/Players/IPlayer/)                  |
-| `items`        | [`List<IEntityItem>`](/Vanilla/Entities/IEntityItem/) |
-| `damageSource` | [IDamageSource](/Vanilla/Damage/IDamageSource/)       |
+| ZenGetter       | ZenSetter       | Type                                              |
+|-----------------|-----------------|---------------------------------------------------|
+| `items`         |  `items`        | [`List<IEntityItem>`](/Vanilla/Entities/IEntityItem/) |
```
Comment (jaredlll08) [inline docs/1.12/content/Vanilla/Events/Events/PlayerDeathDrops.md:21]:
```suggestion
| `items`         |  `items`        | List<[IEntityItem](/Vanilla/Entities/IEntityItem/)> |
```

File: docs/1.12/content/Vanilla/Events/Events/PlayerInteractBlock.md [hunk 17]
```diff
@@ -15,19 +15,53 @@ PlayerInteractBlock Events implement the following interfaces and are able to ca
 
 - [IEventCancelable](/Vanilla/Events/Events/IEventCancelable/)
 - [PlayerInteract](/Vanilla/Events/Events/PlayerInteract/)
-- [IPlayerEvent](/Vanilla/Events/Events/IPlayerEvent/)
+- [IHasCancellationResult] (No doc for it till now)
```
Comment (jaredlll08) [inline docs/1.12/content/Vanilla/Events/Events/PlayerInteractBlock.md:18]:
```suggestion
- IHasCancellationResult
```

**Review by jaredlll08 — CHANGES_REQUESTED** [review]
Thanks for this!

Just a *few* things need changing

File: docs/1.12/content/Vanilla/Events/Events/AnimalTame.md [hunk 18]
```diff
@@ -20,5 +20,14 @@ The following information can be retrieved/set during the event:
...
+
+
+## ZenGetters/ZenSetters/ZenMethods from extensions
+
+| ZenGetter       | ZenSetter       | Type                                              |
+|-----------------|-----------------|---------------------------------------------------|
+| `canceled`      | `canceled`      | bool                                              |
+
+ZenMethodes
+- `event.cancel();` Method, returns void (nothing). Can cancel the event and stop smth. from happening
```
[the remark this answers is not in the document]
Comment (sora7672) [reply docs/1.12/content/Vanilla/Events/Events/AnimalTame.md:33]: Sure, will do. But anywhen later this week. IRL a lil more this week :D

File: docs/1.12/content/Vanilla/Events/Events/BlockNeighborNotify.md [hunk 19]
```diff
@@ -15,9 +15,36 @@ BlockNeighborNotify Events implement the following interfaces and are able to ca
 - [IEventCancelable](/Vanilla/Events/Events/IEventCancelable/)
 - [IBlockEvent](/Vanilla/Events/Events/IBlockEvent/)
 
-## ZenGetters
+## ZenGetters/ZenSetters
+| ZenGetter       | ZenSetter       | Type                                              |
+|-----------------|-----------------|---------------------------------------------------|
+| `forceRedstoneUpdate`  |          | bool                                              |
+| `notifiedSides` |                 | [IFacing](/Vanilla/World/IFacing) [ ]              |
```
[the remark this answers is not in the document]
Comment (sora7672) [reply docs/1.12/content/Vanilla/Events/Events/BlockNeighborNotify.md:22]: Intended with the space between. Better readability on some devices/browsers. Soke show it like a block, which could be confusing

[the remark this answers is not in the document]
Comment (jaredlll08) [reply docs/1.12/content/Vanilla/Events/Events/BlockNeighborNotify.md:22]: I can't imagine that any modern browser would change how plaintext is rendered like that, do you have an example of where it messes up?

[the remark this answers is not in the document]
Comment (sora7672) [reply docs/1.12/content/Vanilla/Events/Events/BlockNeighborNotify.md]: grafik

Looking like this. 
Easyer to read with sapce. But changed all allready ;)
just so you see what i mean with difference.

**Review by jaredlll08 — APPROVED** [review]

**Pull request merged**